JobsPortal/src/main/java/com/ps/js/controller/JobsController.java

```diff
@@ -0,0 +1,363 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ps.js.entity.Department;
+import com.ps.js.entity.JobDetails;
+import com.ps.js.entity.JobLocation;
+import com.ps.js.entity.RolesAndResponsebility;
+import com.ps.js.entity.Skill;
+import com.ps.js.exception.ErrorMessages;
+import com.ps.js.exception.JobDetailsNotFoundException;
+import com.ps.js.exception.LocationsNotFoundException;
+import com.ps.js.exception.RolesAndResponsebilityNotFoundException;
+import com.ps.js.exception.SkillNotFoundException;
+import com.ps.js.mapper.JobDetailsMapper;
+import com.ps.js.payload.JobDetailsPayload;
+import com.ps.js.payload.JobDetailsResponsePayload;
+import com.ps.js.payload.JobLocationPayload;
+import com.ps.js.payload.ResponsePayload;
+import com.ps.js.payload.RolesAndResponsebilityPayload;
+import com.ps.js.payload.SkillPayload;
+
+import com.ps.js.service.IDepartmentService;
+import com.ps.js.service.IJobDetailsService;
+import com.ps.js.service.IJobLocationServices;
+import com.ps.js.service.IRolesAndResponsebilityService;
+import com.ps.js.service.ISkillServices;
+
+@RestController
+@RequestMapping("/jobs")
+public class JobsController {
+
+	/**
+	 * Injecting JobLocationService
+	 */
+	@Autowired
+	private IJobLocationServices jobLocationServices;
+	/**
+	 * Injecting JobDetailsService
+	 */
+	@Autowired
+	private IJobDetailsService jobDetailsService;
+	/**
+	 * Injecting SkillService
+	 */
+	@Autowired
+	private ISkillServices skillServices;
+	/**
+	 * Injecting DepartmentService
+	 */
+	@Autowired
+	private IDepartmentService departmentService;
+	/**
+	 * Injecting rolesAndResponsebilityService
+	 */
+	@Autowired
+	private IRolesAndResponsebilityService rolesAndResponsebilityService;
+	/**
+	 * Injecting JobDetailsMapper
+	 */
+	@Autowired
+	private JobDetailsMapper jobDetailsMapper;
+
+	private static ObjectMapper mapper = new ObjectMapper();
+
+	/**
+	 * For fetching all the jobs currently having in record
+	 * 
+	 * @return {@link ResponseEntity}
+	 */
+
+	@GetMapping("/find-all-job")
+	public ResponseEntity<List<JobDetails>> findAllJob() {
+		// Fetch all the jobs available
+		List<JobDetails> listJobDetails = jobDetailsService.findAllJob();
+		if (listJobDetails == null)
+			throw new JobDetailsNotFoundException(ErrorMessages.NO_JOB_DETAILS_FOUND_EXCEPTION.toString());
+
+		return new ResponseEntity<List<JobDetails>>(listJobDetails, HttpStatus.OK);
+	}
+
+	/**
+	 * For create a new jobDetails and return jobDetails with generated jobId
+	 * 
+	 * @param jobDetailsPayload
+	 * @return {@link ResponseEntity}
+	 */
+	@PostMapping("/open-job")
+	public ResponseEntity<JobDetailsResponsePayload> openJob(@Valid @RequestBody JobDetailsPayload jobDetailsPayload) {
+		Set<JobLocation> locations = new HashSet<JobLocation>();
+		Set<Skill> primarySkills = new HashSet<Skill>();
```

Comment:
Just think once, here we are dealing with very less set of data right? Even when the application is used for a big company with 10Lakh emplyeees also, here these Set variables will hold very less set of data right!
Locations, Skills are limited for a job!
So using Arraylist is Enough